src/world/Actor/Player.cpp

```diff
@@ -603,8 +603,7 @@ void Sapphire::Entity::Player::discover( int16_t map_id, int16_t sub_id )
 
   int32_t offset = 4;
 
-  auto info = exdData.get< Sapphire::Data::Map >(
-    exdData.get< Sapphire::Data::TerritoryType >( getCurrentTerritory()->getTerritoryTypeId() )->map );
+  auto info = exdData.get< Sapphire::Data::Map >( map_id );
```

Comment:
should check that this isn't nullptr before using it otherwise you'll crash the server